tpython/lib/utils/futils.py

```diff
@@ -1,136 +1,129 @@
     if not os.path.exists(folder):
-        os.mkdir(folder)
-    return os.write(folder + '/' + filename, contents)
+        os.makedirs(folder)
+        os.mknod(folder + "/" + filename)
+        file = open(folder + "/" + filename, "a")
+        file.write(contents)
+    else:
+        pass
+
 
 def readFile(path1):
     if os.path.exists(path1):
-        array = os.read(path1).split('\n')
-        return array[0].replace('\r', '')
+        list_of_lines = []
+        open_file = open(path1, "r")
+        read_file = open_file.read()
+        list_of_lines = read_file.splitlines()
+        return list_of_lines[0].replace("\r", "")
     else:
-        logging.warning('Unable to locate ' + path1)
+        logging.warning("Unable to locate " + path1)
+
 
 def getImageFile(name):
-    validImageTypes = ['png','jpg','jpeg','gif']
-    image = ''
+    validImageTypes = ["png", "jpg", "jpeg", "gif"]
+    image = ""
     for type in validImageTypes:
-        image = name + '.' + type
+        image = name + "." + type
         if os.path.exists(image):
             return image
+        else:
+            pass
+
 
 def getParentFolder():
-    return os.path.basename(os.path.dirname(os.getcwd)) 
+    return os.path.basename(os.path.dirname(os.getcwd()))
+
 
 def getDirectories(srcpath):
-    onlyfiles = [f for f in listdir(srcpath) if isfile(join(srcpath, f))]
+    onlyfiles = [f for f in listdir(srcpath) if isdir(join(srcpath, f))]
     return onlyfiles
 
+
 def verifyFolder(folder):
     if not os.path.exists(folder):
-        os.mkdir('-p', folder)
+        os.makedirs(folder, exist_ok=True)
+
 
 def copyFileToFolder(src, dest):
     if os.path.exists(src):
-        os.mkdir('-p', dest)
-        sh.copy('-rf', src, dest)
+        os.makedirs(dest, exist_ok=True)
+        sh.copy2(src, dest)
+
 
 def copyFolder(src, dest):
-    os.mkdir('-p', dest)
-    sh.copy('-rf', src, dest)
+    sh.copytree(src, dest)
 
-def readWholeFile(path1):
-    if os.path.exists(path1):
-        array = os.open(path1)
-        return array
+
+def readWholeFile(path):
+    if os.path.exists(path):
+        open_content = open(path)
+        file_content = open_content.read()
+        return file_content
     else:
-        logging.warning('Unable to locate ' + path1)
+        print("Unable to locate " + path)
+
 
 def readYaml(path):
-    yamldata = ''
-    try:
-        yamldata = yaml.load(os.open(path, encoding='utf-8'))
-    except:
-        logging.warning('Tutors ${version} encountered an error reading properties.yaml:')
-        logging.warning('--------------------------------------------------------------')
-        #logging.warning(err.mark.buffer)
-        logging.warning('--------------------------------------------------------------')
-        #logging.warning(err.message)
-        logging.warning('Review this file and try again....')
+    yamldata = ""
+    with open(path) as yam:
+        yamldata = frontmatter.load(yam)
```

Comment:
might be safer to just `return frontmatter.load(yam)` from within the context manager and then return a none type outside the context manager.